provider/src/main/kotlin/org/gradle/kotlin/dsl/DependencyHandlerScope.kt

```diff
@@ -39,9 +39,76 @@ class DependencyHandlerScope(val dependencies: DependencyHandler) : DependencyHa
     operator fun String.invoke(dependencyNotation: Any): Dependency =
         dependencies.add(this, dependencyNotation)
 
+    /**
+     * Adds a dependency to the given configuration.
+     *
+     * @param dependencyNotation notation for the dependency to be added.
+     * @param dependencyConfiguration expression to use to configure the dependency.
+     * @return The dependency.
+     * @see DependencyHandler.add
+     */
+    inline
     operator fun String.invoke(dependencyNotation: String, dependencyConfiguration: ExternalModuleDependency.() -> Unit): ExternalModuleDependency =
         dependencies.add(this, dependencyNotation, dependencyConfiguration)
 
+    /**
+     * Adds a dependency to the given configuration.
+     *
+     * @param group the group of the module to be added as a dependency.
+     * @param name the name of the module to be added as a dependency.
+     * @param version the optional version of the module to be added as a dependency.
+     * @param configuration the optional configuration of the module to be added as a dependency.
+     * @param classifier the optional classifier of the module artifact to be added as a dependency.
+     * @param ext the optional extension of the module artifact to be added as a dependency.
+     * @return The dependency.
+     *
+     * @see DependencyHandler.add
+     */
+    operator fun String.invoke(
+        group: String,
+        name: String,
+        version: String? = null,
+        configuration: String? = null,
+        classifier: String? = null,
+        ext: String? = null): ExternalModuleDependency =
+        dependencies.create(group, name, version, configuration, classifier, ext).apply { add(this@invoke, this) }
+
+    /**
+     * Adds a dependency to the given configuration.
+     *
+     * @param group the group of the module to be added as a dependency.
+     * @param name the name of the module to be added as a dependency.
+     * @param version the optional version of the module to be added as a dependency.
+     * @param configuration the optional configuration of the module to be added as a dependency.
+     * @param classifier the optional classifier of the module artifact to be added as a dependency.
+     * @param ext the optional extension of the module artifact to be added as a dependency.
+     * @param dependencyConfiguration expression to use to configure the dependency.
+     * @return The dependency.
+     *
+     * @see DependencyHandler.create
+     * @see DependencyHandler.add
+     */
+    inline
+    operator fun String.invoke(
+        group: String,
+        name: String,
+        version: String? = null,
+        configuration: String? = null,
+        classifier: String? = null,
+        ext: String? = null,
+        dependencyConfiguration: ExternalModuleDependency.() -> Unit): ExternalModuleDependency =
```

Comment:
Why do you need the overload above? Why not just make the default value for this param `{ }`?
Also, shouldn't this param be marked `noinline`? I'm surprised the compiler isn't blowing up on you for not having that keyword.

Reply:
Good point, I unified the extensions according to what we generate for discovered configurations. I'll have a closer look, maybe we can reduce the number of extensions.

Reply:
This wasn't possible at the time this overload was introduced. Makes sense to add a default value to the argument (assuming all current usages of the method are supported).

Reply:
`e: ../org/gradle/kotlin/dsl/DependencyHandlerScope.kt: (99, 9): 'dependencyConfiguration: ExternalModuleDependency.() -> Unit = {}' construction is not yet supported in inline functions`

So, I'll keep the overloads for now.